File: algorithms/M/Minimum Time Visiting All Points/Minimum Time Visiting All Points.cpp
class Solution {
public:
    int minTimeToVisitAllPoints(vector<vector<int>>& points) {
        vector<int>sk;
        int x,y,maxy=0;
        for(int i=0;i<points.size()-1;i++){
            for(int j=0;j<points[i].size()-1;j++){
                x=abs(points[i][j]-points[i+1][j]);
                y=abs(points[i][j+1]-points[i+1][j+1]);
                maxy+=std::max(x,y);
            }
        }
        return maxy;
    }
};

File: scripts/algorithms/M/Minimum Time Visiting All Points/Minimum Time Visiting All Points.py
# Runtime: 38 ms (Top 57.7%) | Memory: 13.27 MB (Top 67.6%)

class Solution:
    def minTimeToVisitAllPoints(self, points):
        res = 0
        n = len(points)
        for i in range(n-1):
            dx = abs(points[i+1][0]-points[i][0])
            dy = abs(points[i+1][1]-points[i][1])
            res+= max(dx,dy)
        return res

            
obj = Solution()
print(obj.minTimeToVisitAllPoints([[1,1],[3,4],[-1,0]]))